streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/AssignorConfigurationTest.java

```diff
@@ -35,4 +53,52 @@ public void configsShouldRejectZeroWarmups() {
 
         assertThat(exception.getMessage(), containsString("Invalid value 0 for configuration max.warmup.replicas"));
     }
+
+    @Test
+    public void rebalanceProtocolShouldSupportAllUpgradeFromVersions() {
+        for (final String upgradeFrom : TestConfig.upgradeFromVersions()) {
+            config.put(StreamsConfig.UPGRADE_FROM_CONFIG, upgradeFrom);
+            final AssignorConfiguration assignorConfiguration = new AssignorConfiguration(config);
+
+            // next call should not throw
+            assignorConfiguration.rebalanceProtocol();
```

Comment:
A more readable option without the need for a comment is:
```suggestion
            try {
                assignorConfiguration.rebalanceProtocol(); 
            } catch (final Throwable throwable) {
                throw new AssertionFailedError("Upgrade from " + upgradeFrom + " failed with " + throwable.getMessage() + "!");
            }       
```